testing/preview/exercise-preview-page.tsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable import/no-relative-packages, import/order */
+import {View} from "@khanacademy/wonder-blocks-core";
+import {spacing} from "@khanacademy/wonder-blocks-tokens";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+
+import {PreviewRenderer} from "./preview-renderer";
+
+import type {PreviewContent} from "../../packages/perseus-editor/src/preview/message-types";
+
+/**
+ * Loads the exercise preview frame
+ *
+ * This is loaded inside the iframe, where it sets up the ExercisePreviewFrame
+ * component that handles all communication between the iframe and its parent.
+ */
+const ExercisePreviewPage = () => {
+    const [previewData, setPreviewData] = React.useState<PreviewContent | null>(
+        null,
+    );
+    const [iframeId, setIframeId] = React.useState<string | null>(null);
+    const containerRef = React.useRef<HTMLDivElement>(null);
+
+    // Read iframe configuration from dataset attributes
+    React.useEffect(() => {
+        const iframe = window.frameElement as HTMLIFrameElement | null;
+        if (iframe) {
+            const id = iframe.dataset.id;
+            if (id) {
+                setIframeId(id);
+            }
+        }
+    }, []);
+
+    // Listen for data from parent
+    React.useEffect(() => {
+        const handleMessage = (event: MessageEvent) => {
+            // For now, we receive the iframe ID and look up data in parent's iframeDataStore
+            if (
+                typeof event.data === "string" ||
+                typeof event.data === "number"
+            ) {
+                const id = String(event.data);
+                // Access parent's iframeDataStore
+                try {
+                    // @ts-expect-error - accessing parent window's custom property
+                    const data = window.parent.iframeDataStore?.[id];
+                    if (data) {
+                        setPreviewData(data);
+                    }
+                } catch (e) {
+                    // Cross-origin access might fail in some cases
+                    // Silently ignore - this is expected in cross-origin scenarios
```

Comment:
Should we at least log the error here? Or does the browser log something for us?